src/sentry/utils/http.py

```diff
@@ -20,14 +29,14 @@ def absolute_uri(url: Optional[str] = None, url_prefix=None) -> str:
     return urljoin(url_prefix.rstrip("/") + "/", url.lstrip("/"))
 
 
-def origin_from_url(url):
+def origin_from_url(url: str) -> str:
     if not url:
         return url
-    url = urlparse(url)
-    return f"{url.scheme}://{url.netloc}"
+    parsed = urlparse(url)
```

Comment:
Good catch! Do not touch those input variables! :)

Reply:
heh yeah mypy screamed!